@Sim/initializeUASTrajTch.m

```diff
@@ -0,0 +1,212 @@
+      t_i = 0;
+      while ~ok && t_i < max_iter
+        %                         x0 = [minx+(maxx-minx)*rand(),miny+(maxy-miny)*rand()];
+        %                         xf = [minx+(maxx-minx)*rand(),miny+(maxy-miny)*rand()];
+        % Create a trajectory based on this UAS capabilities
+        [traj, lane_ids, vert_ids, toa_s, ok] = ...
+          uas_i.createTrajectoryRandVerts(obj.sim_config.fit_traj);
+        t_i = t_i + 1;
+      end
+      if ~ok
+        error("Failed to create proper trajectory");
+      end
+    else
+      l = obj.sim_config.single_lane; %this was for testing single lane
+      [traj, lane_ids, vert_ids, toa_s] = ...
+        uas_i.createTrajectoryLane(l, obj.sim_config.fit_traj);
+    end
+    % Generate a random request time
+
+    %I dont think request time is possible in rogue - this
+    %causes discrepancy between the two algos so I set it to t0
+    %r = t0+(tf-t0)*rand();
+    r = t0;
+    uas_i.r_s = r;
+    % Move the time-of-arrival of the trajectory to this time
+    toa_s = toa_s+r;
+    % The earliest and latest possible release
+    % For renyi set to the request time
+    r_t0 = max(r-uas_i.flex, t0);
+    r_tf = min(r+uas_i.flex, tf);
+
+    lane_lengths = obj.lbsd.getLaneLengths(lane_ids);
+    reservations = obj.lbsd.getReservations();
+
+    reservations_rogue = [];
+    [count_lanes, ~] = size(obj.lbsd.lane_graph.Edges);
+
+    path = [];
+
+    for k = 1:count_lanes
+      reservations_rogue(k).flights= [];
+    end
+
+    %somehow here I could combine reservations
+    for k = 1:height(reservations)
+      lane = str2num(reservations(k, :).lane_id);
+      current_flights = reservations_rogue(lane).flights;
+      current_flights = [current_flights;
+        -1, reservations(k, :).entry_time_s, reservations(k, :).exit_time_s, reservations(k, :).speed ];
+      reservations_rogue(lane).flights = current_flights;
+    end
+
+    for k = 1:size(lane_ids)
+      path(k) = str2num(lane_ids(k));
+    end
+
+
+    speed = uas_i.nominal_speed;
+
+
+
+    %this algorithm for deconfliction is different than LSD but could be used - see
+    %LEM_test_res for comparison between below func and the
+    %original reserveLBSDTrajectory
+    %t_start = LBSD.LEM_launch_time_nc(obj,reservations,path,t1,t2,lane_lengths,...
+    %hd,speed);
+
+
+    timerVal = tic;
+
+
+    [flight_plan,reservations_rogue] = obj.lbsd.LEM_reserve_fp_rogue( ...
+      reservations_rogue, lane_lengths, r, r_tf, speed,path,uas_i.h_d);
+
+
+    res_ids_rogue = [];
+    res_toa_s_rogue = [];
+
+    res_times(i) = toc(timerVal);
+
+
+    % Reserve the trajectory
+    %timerVal = tic;
+    [ok, res_ids, res_toa_s] = ...
+      obj.lbsd.reserveLBSDTrajectory(lane_ids, uas_i.id, toa_s, ...
+      uas_i.h_d, r_t0, r_tf);
+    %res_times(i) = toc(timerVal);
+
+    %todo will flight plan be empty if can't be scheduled?
+    %when I was running Tom's code it would throw and error and
+    %break
+    if ~isempty(flight_plan)
```

Comment:
Ideally all this stuff would be in the LBSD class so we can call reserveLBSDTrajectoryTch (or something like that) and it will have the same signature as reserveLBSDTrajectory